WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -19,7 +20,42 @@ def new
   end
 
   def index
-    @competitions = Competition.all.select([:id, :name, :cityName, :countryId]).order(:year, :month, :day).reverse_order
+    @regions = { '' => [["All","all"]],
+                 'Continent' => Continent.all.map { |continent| [continent.name, continent.id] },
+                 'Country' => Country.all.map { |country| [country.name, country.id] } }
+    @events = [ ["All", "all"], ["",""] ] + Event.all_official.map { |event| [event.name, event.id] }
+    @years = [ ["Current","current"],["All","all"],["",""] ] + Competition.select(:year).map(&:year).uniq.reverse!
```

Comment:
I think the space you put here between "All" and the rest of the events is nice, but, IMO, it should be disabled, so people cannot select it. See http://makandracards.com/makandra/17525-rails-disable-options-of-a-select-field for how to disabled certain options of a select in rails.